scripts/rqtl2_wrapper.R

```diff
@@ -0,0 +1,447 @@
+ make_option(c("-d", "--directory"), action = "store", default = NULL, type = "character", help="Temporary working directory: should also host the input file ."),
+ make_option(c("-m", "--method"), action = "store", default = "HK", type = "character", help="Scan Mapping Method - HK (Haley Knott), LMM( Linear Mixed Model ), LOCO (Leave one Chromosome Out)"),
+make_option(c("-o", "--output_file"), action="store", default=NULL, type='character', help="a file name of where to write the output json results"),
+  make_option(c("--pstrata"), action="store_true", default=NULL, help="Use permutation strata")
+)
+
+
+opt <- parse_args(OptionParser(option_list=option_list))
+# check for mandatory args
+if(is.null(opt$directory)){
+stop("You need  to specify  a working temporary directory which should have be base_dir for the input file")
+}
+
+NO_OF_CORES = opt$cores
+SCAN_METHOD = opt$method
+NO_OF_PERMUTATION = opt$nperm
+
+# get the json file path with pre metadata required to create the cross
+# assumption is that the input file should be in the  temp working directory
+
+if (is.null(opt$input_file)) {
+  stop("Argument for the Input metadata file is Missing ", call. = FALSE)
+} else {
+   input_file = opt$input_file
+}
+if (is.null(opt$output_file)){
+stop("You need to provide an output file to write the ouput data")
+}
+
+# file_path
+
+input_file_path = file.path(opt$directory, input_file)
+
+if (!(file.exists(input_file_path))) {
+  str_glue("The input file {opt$input_file} path does not exists in the directory {opt$directory}")
+  stop("The input file does not exists the temp directory")
+} else {
+  str_glue("The input path for the metadata >>>>>>> {input_file_path}")
+  json_data  = fromJSON(file = input_file_path)
+}
+
+# generate random string file paths
+
+genRandomFileName <- function(prefix, file_ext = ".txt") {
+  randStr = paste(prefix, stri_rand_strings(1, 9, pattern = "[A-Za-z0-9]"), sep =
+                    "_")
+  return(paste(randStr, file_ext, sep = ""))
+}
+
+# TODO work on the optional parameters e.g cores, type of computation
+
+# TODO create temp directory for this workspace pass this as argument
+
+control_file_path  <- file.path(opt$directory,
+                                genRandomFileName(prefix = "control_", file_ext = ".json"))
+
+str_glue("Generated control file path is  {control_file_path}")
+
+if (is.null(json_data$sep)){
+cat("Using ',' as a default sep for cross file\n")
+json_data$sep = ","
+}
+if (is.null(json_data$na.strings)){
+cat("Using '-' and 'NA' as the default na.strings\n")
+ json_data$na.strings = c("-" , "NA")
+}
+
+# use this better defaults
+default_keys = c(
+                "geno_transposed", "founder_geno_transposed",
+                "pheno_transposed" , "covar_transposed",
+		"phenocovar_transposed")
+
+for (item in default_keys) {
+if (!(item %in% names(json_data))){
+  cat("Using FALSE as default parameter for ", item)
```

Comment:
minor: would this still work: `cat("Using FALSE as default parameter for ", item, "\n")` removing the need for `cat("\n")`